File: SpringBoot-Simple-JPA/src/main/java/com/example/springbootsimplejpa/controller/OrderController.java
package com.example.springbootsimplejpa.controller;


import com.example.springbootsimplejpa.domain.OrderModel;
import com.example.springbootsimplejpa.service.OrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/order")
public class OrderController {


    @Autowired
    private OrderService orderService;


    @GetMapping
    public List<OrderModel> findAllOrderController() {
        return orderService.getAllOrder();
    }

    @PostMapping
    public OrderModel saveOrderController(@RequestBody OrderModel order) {
        return orderService.saveOrder(order);
    }


}

File: SpringBoot-Multimodules/module-api/src/main/kotlin/com/wool/controller/OrderController.kt
package com.wool.controller

import com.wool.service.OrderService
import com.wool.service.dtos.order.OrderRequestDto
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController


@RestController
class OrderController(
    private val orderService: OrderService
) {

    @GetMapping("/orders")
    fun getOrders() = orderService.getOrders()

    @PostMapping("/orders")
    fun saveOrders(@RequestBody orderRequestDto: OrderRequestDto) {
        //save order
        return orderService.saveOrder(orderRequestDto)
    }
}
